### PR TITLE
Observe Y.XmlText values, don't allow empty title (or empty slug for …

### PR DESCRIPTION
**Pull request by dannelundqvist**
…text assignment)

### CODE REVIEW

File: src/hooks/useYValue.tsx [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+ * @param deep boolean
+ *
+ * @returns [<T>, YParent]
+ */
+export function useYValue<T>(path: string, deep: boolean = false): [
+  T | undefined,
+  YParent
+] {
+  const { provider, synced } = useCollaboration()
+  const yRoot = synced ? provider?.document.getMap('ele') : undefined
```
Comment (gularsson) [inline src/hooks/useYValue.tsx:24]: There was something about synced... It only allows server changes, not offline...?

**Review by gularsson — APPROVED** [review]
Great work, I feel that we could probably replace the useYObserver for this one eventually. :muscle:

**Comment by dannelundqvist** [comment]
Also implemented setter so that the useYValue() hook now works like useState(). Regardless of whether the values parent container is a Y.Map or Y.Array.

Example usage:

```javascript
const [assignmentType, setAssignmentType] = useYValue<string | undefined>(`meta.core/assignment[${index}].meta.core/assignment-type[0].value`)
...
setAssignmentType('graphic')
```

**Review by gularsson — APPROVED** [review]
:+1:

**Pull request merged**